File: backend/GofishApi/Dtos/GetFeedDTO.cs
using GofishApi.Enums;

namespace GofishApi.Dtos;

public record GetFeedReqDTO(
    FeedKind Kind,
    PostDataRequestDTO? DataRequest,
    DateTime LastTimestamp,
    int MaxResults
)
{ }

public record GetFeedResDTO(
    IReadOnlyCollection<GetPostsPostDTO> Posts,
    bool HasMoreResults,
    DateTime? LastTimestamp
)
{ }

File: angular/src/app/shared/dtos/get-feed.dto.ts
import { PostDataRequestDTO, GetPostsPostDTO } from './get-post.dto';

export interface GetFeedReqDTO {
  kind: number;
  dataRequest?: PostDataRequestDTO;
  lastTimestamp: string;
  maxResults: number;
}

export interface GetFeedResDTO {
  posts: GetPostsPostDTO[];
  hasMoreResults: boolean;
  lastTimestamp?: string;
}
